extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/SyntheticBeanBuildItem.java

```diff
@@ -32,6 +32,9 @@
 public final class SyntheticBeanBuildItem extends MultiBuildItem {
 
     /**
+     * Returns a configurator object allowing for further customization of the synthetic bean.
+     * <p>
+     * The implementation class is automatically registered as a resulting bean type.
```

Comment:
@mkouba @Ladicek I did not `@Deprecate` this yet, should we do that?
While I would prefer to have a single way of registering the bean here, I can see use in both approaches.

Reply:
Agreed, it seems to me that both cases have value.

Reply:
Ok, let's keep it then.